src/http-handler.js

```diff
@@ -0,0 +1,145 @@
+import debugFactory from 'debug';
+import getRawBody from 'raw-body';
+import qs from 'qs';
+import crypto from 'crypto';
+import { packageIdentifier } from './util';
+
+export const errorCodes = {
+  SIGNATURE_VERIFICATION_FAILURE: 'SLACKMESSAGEMIDDLEWARE_REQUEST_SIGNATURE_VERIFICATION_FAILURE',
+  REQUEST_TIME_FAILURE: 'SLACKMESSAGEMIDDLEWARE_REQUEST_TIMELIMIT_FAILURE',
+  BODY_PARSING_FAILED: 'SLACKMESSAGEMIDDLEWARE_BODY_PARSING_FAILURE',
+  BODY_PARSER_NOT_PERMITTED: 'SLACKMESSAGEMIDDLEWARE_BODY_PARSER_NOT_PERMITTED_FAILURE',
+};
+
+const debug = debugFactory('@slack/interactive-messages:http-handler');
+
+export function createHTTPHandler(adapter) {
+  const poweredBy = packageIdentifier();
+
+  /**
+   * Parses raw bodies of requests
+   *
+   * @param {Object} res - Response object
+   * @returns {Function} Returns a function used to send response
+   */
+  function sendResponse(res) {
+    return function _sendResponse(dispatchResult) {
+      const { status, content } = dispatchResult;
+      res.statusCode = status;
+      res.setHeader('X-Slack-Powered-By', poweredBy);
+      if (typeof content === 'string') {
+        res.end(content);
+      } else if (content) {
+        res.end(JSON.stringify(content));
+      } else {
+        res.end();
+      }
+    };
+  }
+
+  /**
+   * Parses raw bodies of requests
+   *
+   * @param {Object} req - Request object
+   * @param {string} body - Raw body of request
+   * @returns {Object} Parsed body of the request
+   */
+  function parseBody(req, body) {
+    const type = req.headers['content-type'];
+    let parsedBody;
+
+    if (type === 'application/x-www-form-urlencoded') {
```

Comment:
we may not need to do this content-type checking if we know that all interactive message, dialog submission/cancellation, and action requests are always urlencoded (never JSON). we need to double-check this.